### ISSUE TITLE
Spring webflux, Mongo reactive connection leak while query by flux ids with cancel request manual

### ISSUE THREAD

**Issue by efsn** [spring-projects/spring-boot#17821]
I have a method which with the flux ids paramter. 
```
fun findXxByIdIn(ids: Flux<String>): Flux<Xx>
```
Now I boot my server which use the webflux for http request, when I request a GET request which will tune the `findXxByIdIn` method, before return the query result, I cancel my http request, repeat about 10 times, then my server will process any request, likeness play dead.

**Comment by efsn** [comment]
However list is OK!
```
fun findXxByIdIn(ids: List<String>): Flux<Xx>
```

**Comment by snicoll** [comment]
@efsn it's hard to say what is going on with the information you've shared as it could be a problem with the infrastructure, your code or a library (not necessarily Spring Boot). To be sure, please share a small sample (github repo or zip) that we can use to reproduce the problem ourselves.

**Comment by efsn** [comment]
Hi, my code is a really spring boot project with the version below, if you need a sample then I will share you late as soon as possible.
```
id("org.springframework.boot") version "2.1.6.RELEASE"
api("org.springframework.boot:spring-boot-starter-webflux")
```

**Comment by snicoll** [comment]
Yes we need a sample and instructions how to reproduce the problem.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.
